src/common/LwrServerUtils.ts

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { CommonUtils } from '@salesforce/lwc-dev-mobile-core/lib/common/CommonUtils';
+import { createServer, LwrApp } from 'lwr';
+import { LwrGlobalConfig } from '@lwrjs/types';
+import {
+    DirModuleRecord,
+    LwrRoute,
+    ServiceEntry
+} from '@lwrjs/types/build/config';
+
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+
+const defaultServerTimeout = 30 * 60 * 1000; // 30 minutes
+
+export class LwrServerUtils {
+    public static async startLwrServer(
+        componentName: string,
+        projectDir: string,
+        idleTimeout: number = defaultServerTimeout,
+        exitOnIdle: boolean = true
+    ): Promise<string> {
+        const lwrApp = createServer(
+            LwrServerUtils.getMergedLwrConfig(componentName, projectDir)
+        );
+
+        const runtimeConfig = lwrApp.getConfig();
+
+        return lwrApp
+            .listen(() => {
+                // tslint:disable-next-line: no-console
+                console.log(
+                    `Listening on port ${runtimeConfig.port} ( mode = ${runtimeConfig.serverMode} , type = ${runtimeConfig.serverType} )`
+                );
+                LwrServerUtils.setServerIdleTimeout(lwrApp, idleTimeout, () => {
+                    if (exitOnIdle) {
+                        process.exit(0); // kill the process on server timeout
+                    }
```

Comment:
When running the product code we do want to exit the process but when running unit tests we don't want that to happen. The simplest/cleanest way to make this overridable was to add `exitOnIdle` parameter to `startLwrServer` function.

Reply:
Why do we want the process to keep running for unit test?

Reply:
B/c exiting the process will throw off jest test runner. Jest test runner will exit the test process by itself after the test is done running and if it does not have any handles/ports left open. If it does have handle/ports left open then jest will print out a warning about that (which is a great way to ensure that our processes clean up after themselves). The fact that our tests runs to the end and jest test running doesn't print out any warning at the end means that we are properly shutting down the server when we detect it being idle.